scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -446,228 +460,232 @@
-                                          ; Who resurrected this pod? Where did it come from? Do we have two instances of cook?
-                                          :pod/running (kill-pod-in-weird-state compute-cluster pod-name
-                                                                                cook-expected-state-dict
-                                                                                k8s-actual-state-dict)
-                                          ; The writeback to datomic has occurred, so there's nothing to do except to delete the pod from kubernetes
-                                          ; and remove it from our tracking.
-                                          :pod/succeeded (delete-pod api-client name cook-expected-state-dict pod)
-                                          ; TODO: Should mark mea culpa retry
-                                          :pod/unknown (handle-pod-completed-and-kill-pod-in-weird-state
-                                                         compute-cluster pod-name k8s-actual-state-dict)
-                                          ; Who resurrected this pod? Where did it come from? Do we have two instances of cook?
-                                          :pod/waiting (kill-pod-in-weird-state compute-cluster pod-name
-                                                                                cook-expected-state-dict
-                                                                                k8s-actual-state-dict))
-
-                                        :cook-expected-state/killed
-                                        (case pod-synthesized-state-modified
-                                          :missing
-                                          ; There's a race where we can launch then kill, when the kill arrives after
-                                          ; the launch, but before the watch updates k8s-actual-state.
-                                          ; (k8s-actual-state isn't the actual state, because of this watch lag)
-                                          ; When that happens, we will have an actual state of :missing.
-                                          ; If we did nothing, we'd log this as a weird state, then when the watch
-                                          ; shows up, we'd see (:missing, :starting) and log that as a weird state too.
-                                          ;
-                                          ; So, a better approach. If we detect a (:killed, :missing), then we opportunistically
-                                          ; try to kill the pod. This is why update-cook-expected-state saves :launch-pod,
-                                          ; so its available here.
-                                          (if-let [pod (some-> cook-expected-state-dict :launch-pod :pod)]
+  (prom/with-duration
+    prom/controller-process-duration {:compute-cluster name}
```

Comment:
Lets include the `:doing-scan?` flag here. That will help distinguish the metrics form scans from normal processing of actions. (And let us remove the 'boring percentiles' that scan introduced.

Reply:
Updated!